### PR TITLE
Fix shape in zarr.json in zarr3 streaming

### PR DESCRIPTION
**Pull request by MichaelBuessemeyer**
Fixes shape field in returned `zarr.json` by zarr 3 streaming. 

### URL of deployed dev instance (used for testing):
- https://___.webknossos.xyz

### Steps to test:
- https://fixzarrstreamingshape.webknossos.xyz/data/zarr3_experimental/sample_organization/l4_sample/color/1/zarr.json
- https://fixzarrstreamingshape.webknossos.xyz/data/zarr3_experimental/sample_organization/l4_sample/color/2-2-1/zarr.json
- https://fixzarrstreamingshape.webknossos.xyz/data/zarr3_experimental/sample_organization/l4_sample/color/4-4-1/zarr.json
- https://fixzarrstreamingshape.webknossos.xyz/data/zarr3_experimental/sample_organization/l4_sample/color/8-8-2/zarr.json
- The above should show correctly decreasing shapes.

### TODOs:
- [x] An optional TODO would be to reduce code duplication between `Zarr3ArrayHeader.scala` and its zarr version 2 equivalent.
  - [x] wont be done here 

### Issues:
- contributes to https://github.com/will-moore/ome2024-ngff-challenge/pull/1/#issuecomment-2382906727

------
- [ ] Updated [changelog](../blob/master/CHANGELOG.unreleased.md#unreleased)
- [x] Needs datastore update after deployment

### CODE REVIEW

File: webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/zarr3/Zarr3ArrayHeader.scala [hunk 1]
```diff
@@ -251,13 +252,20 @@ object Zarr3ArrayHeader extends JsonImplicits {
       )
 
   }
-  def fromDataLayer(dataLayer: DataLayer): Zarr3ArrayHeader = {
+  def fromDataLayer(dataLayer: DataLayer, mag: Vec3Int): Zarr3ArrayHeader = {
     val additionalAxes = reorderAdditionalAxes(dataLayer.additionalAxes.getOrElse(Seq.empty))
+    // TODOM: Make code together with ZarrHeader.scala DRYer
```
Comment (MichaelBuessemeyer) [inline webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/zarr3/Zarr3ArrayHeader.scala:257]: I copied over this code from `ZarrHeader.scala`. @fm3 Do you think it makes sense to invest some time to extracted shared code from both files to make the code DRYer or do you think it is ok as it is now (and I just remove this comment here)?

Comment (fm3) [reply webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/zarr3/Zarr3ArrayHeader.scala:257]: I think it’s fine

Comment (MichaelBuessemeyer) [reply webknossos-datastore/app/com/scalableminds/webknossos/datastore/datareaders/zarr3/Zarr3ArrayHeader.scala]: Ok I just removed the comment :)

File: webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/volume/Zarr3BucketStreamSink.scala [hunk 2]
```diff
@@ -41,7 +41,7 @@ class Zarr3BucketStreamSink(val layer: VolumeTracingLayer, tracingHasFallbackLay
   def apply(bucketStream: Iterator[(BucketPosition, Array[Byte])], mags: Seq[Vec3Int], voxelSize: Option[VoxelSize])(
       implicit ec: ExecutionContext): Iterator[NamedStream] = {
 
-    val header = Zarr3ArrayHeader.fromDataLayer(layer)
+    val header = Zarr3ArrayHeader.fromDataLayer(layer, mags.head)
```
Comment (fm3) [inline webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/volume/Zarr3BucketStreamSink.scala:44]:
```suggestion
    val header = Zarr3ArrayHeader.fromDataLayer(layer, mags.headOption.getOrElse(Vec3Int.ones))
```
`head` throws exceptions on empty lists. Not sure how to create volume annotations with no mags, but I feel more comfortable with this variant.

Comment (MichaelBuessemeyer) [reply webknossos-tracingstore/app/com/scalableminds/webknossos/tracingstore/tracings/volume/Zarr3BucketStreamSink.scala:44]: sure, thanks for the suggestion :pray:

**Review by fm3 — APPROVED** [review]
Nice! With a changelog entry, this should be good to go :)

File: CHANGELOG.unreleased.md [hunk 3]
```diff
@@ -22,6 +22,7 @@ For upgrade instructions, please check the [migration guide](MIGRATIONS.released
 
 ### Fixed
 - Fixed a bug during dataset upload in case the configured `datastore.baseFolder` is an absolute path. [#8098](https://github.com/scalableminds/webknossos/pull/8098) [#8103](https://github.com/scalableminds/webknossos/pull/8103)
+- Fixed a bug where the materialize volume annotation job would not start due to a missing parameter. For now the name of the output segmentation layer of the job is autogenerated. [#8116](https://github.com/scalableminds/webknossos/pull/8116)
```
Comment (fm3) [inline CHANGELOG.unreleased.md:25]: I think this text matches #8118 instead ;)

Comment (MichaelBuessemeyer) [reply CHANGELOG.unreleased.md:25]: :woozy_face: Thanks for finding this :pray:

Comment (MichaelBuessemeyer) [reply CHANGELOG.unreleased.md]: Definitely too much pr switcheroo today :see_no_evil:

**Pull request merged**